obspy/core/frequency_domain_trace.py

```diff
@@ -88,9 +86,50 @@ def cross_correlation(self, tr2):
     def frequencies(self):
         n = self.data.size
         timestep = self.stats.delta
-        freq = fft.fftfreq(n, d=timestep)       
+        freq = fft.rfftfreq(n, d=timestep)       
         return freq
 
+    def cross_spectrum(self, tr2):
+        data1 = self.data
+        data2 = tr2.data
+        data2_conj = np.conj(data2)
+        cross_spectrum = data2_conj*data1
+        return (cross_spectrum)
+
+    def coherence(self, tr2):
+        data1 = self.data
+        data2 = tr2.data
+        data2_conj = np.conj(data2)
+        cross_spectrum = data2_conj*data1
+        coherence = (cross_spectrum)**2/(np.abs(data1)*np.abs(data2))
```

Comment:
@fstettner, smoothing of spectra is missing here..

In the long run I would propose adding a method to FFTTrace, e.g. `def smoothed(self, window, npts)` (or width given in Hz).
For now you could just hardcode some fixed window from scipy, see http://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.get_window.html#scipy.signal.get_window. For orientation, Poupinet 1984 used a five point triangular window for 100Hz sampled data.

CC @jwassermann